xrmocap/human_perception/keypoints_estimation/mediapipe_estimator.py

```diff
@@ -0,0 +1,246 @@
+import cv2
+import logging
+import mediapipe as mp
+import numpy as np
+from tqdm import tqdm
+from typing import Tuple, Union
+from xrprimer.utils.ffmpeg_utils import video_to_array
+from xrprimer.utils.log_utils import get_logger
+
+from xrmocap.transform.convention.keypoints_convention import get_keypoint_num
+from .mmpose_top_down_estimator import MMposeTopDownEstimator
+
+
+class MediapipeEstimator(MMposeTopDownEstimator):
+
+    def __init__(self,
+                 mediapipe_kwargs: dict,
+                 bbox_thr: float = 0.0,
+                 logger: Union[None, str, logging.Logger] = None) -> None:
+        """Init a detector from mediapipe.
+
+        Args:
+            mediapipe_kwargs (dict):
+                A dict contains args of mediapipe.
+                refer to https://google.github.io/mediapipe/solutions/pose.html
+                in detail.
+            bbox_thr (float, optional):
+                Threshold of a bbox. Those have lower scores will be ignored.
+                Defaults to 0.0.
+            logger (Union[None, str, logging.Logger], optional):
+                Logger for logging. If None, root logger will be selected.
+                Defaults to None.
+        """
+        # build the pose model
+        mp_pose = mp.solutions.pose
+        self.pose_model = mp_pose.Pose(**mediapipe_kwargs)
+        self.bbox_thr = bbox_thr
+        self.logger = get_logger(logger)
+        self.convention = 'mediapipe_body'
+
+    def get_keypoints_convention_name(self) -> str:
+        """Get data_source from dataset type of the pose model.
+
+        Returns:
+            str:
+                Name of the keypoints convention. Must be
+                a key of KEYPOINTS_FACTORY.
+        """
+        return self.convention
+
+    def infer_single_img(self, img_arr: np.ndarray, bbox_list: list):
+        multi_kps2d = []
+        for bbox_dict in bbox_list:
+            bbox = bbox_dict['bbox']
+            kps2d = None
+            if bbox[4] > self.bbox_thr:
+                img = img_arr[int(bbox[1]):int(bbox[3]),
+                              int(bbox[0]):int(bbox[2])]
+                result_mp = self.pose_model.process(
+                    cv2.cvtColor(img, cv2.COLOR_BGR2RGB))
+                if result_mp.pose_landmarks:
+                    kps_list = [[
+                        landmark.x * img.shape[1] + bbox[0],
+                        landmark.y * img.shape[0] + bbox[1],
+                        landmark.visibility
+                    ] for landmark in result_mp.pose_landmarks.landmark]
+                    kps2d = np.array(kps_list)
+            if kps2d is not None:
+                multi_kps2d.append(dict(bbox=bbox, keypoints=kps2d))
+        return multi_kps2d
```

Comment:
Same as https://github.com/openxrlab/xrmocap/pull/60/files#r1051854721. It returns 2 values, one is kps list, the other is bbox list.   
If some object's bbox has no kps2d, the bbox won't be returned.